File: src/main/java/ru/ifmo/is/mfl/reviews/dto/ReviewDto.java
package ru.ifmo.is.mfl.reviews.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import ru.ifmo.is.mfl.common.framework.dto.CrudDto;
import ru.ifmo.is.mfl.movies.dto.MovieDto;
import ru.ifmo.is.mfl.users.dto.UserDto;

import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = true)
public class ReviewDto extends CrudDto {
  private int id;
  private UserDto user;
  private MovieDto movie;
  private boolean visible;
  private String text;
  private int rating;
  private Instant date;
  private int viewedCounter;
}

File: frontend/src/interfaces/review/dto/ReviewDto.ts
import { CrudDto } from "@/interfaces/crud/dto/CrudDto";
import { MovieDto } from "@/interfaces/movie/dto/MovieDto";
import { UserDto } from "@/interfaces/user/dto/UserDto";
import { DateTime } from "luxon";

export interface ReviewDto extends CrudDto {
  user: UserDto;
  movie: MovieDto;
  visible: boolean;
  text: string;
  rating: number; // 1-10
  date: DateTime
  viewedCounter: number;
}